plugins/fabric8-cdn-support/src/test/java/com/redhat/che/cdn/HttpYamlDownloaderTest.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (c) 2016-2018 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package com.redhat.che.cdn;
+
+import static org.testng.Assert.assertEquals;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import org.eclipse.che.api.workspace.server.wsplugins.model.PluginMeta;
+import org.testng.annotations.Test;
+
+public class HttpYamlDownloaderTest {
+  private final String EDITOR_PLUGIN_URL =
+      "https://che-plugin-registry.openshift.io/v2/plugins/eclipse/che-theia/next/";
+  private final String EDITOR_PLUGIN_VERSION = "next";
+  private final String EDITOR_PLUGIN_REPOSITORY = "https://github.com/eclipse/che-theia";
+  private final String EDITOR_PLUGIN_CATEGORY = "Editor";
+
+  @Test
+  public void getYamlResponseAndParse() throws URISyntaxException, IOException {
+    HttpYamlDownloader httpYamlDownloader = new HttpYamlDownloader();
+    PluginMeta pluginMeta = httpYamlDownloader.getYamlResponseAndParse(new URI(EDITOR_PLUGIN_URL));
```

Comment:
I don't think coding network requests into test cases is a good idea (and is the reason I didn't include any tests to begin with). This could potentially be very fragile.

Reply:
why do you think having integration tests is a bad idea? this could be broken every now and than due to changes in the registry but we would catch it immediately and fail fast IMO

Reply:
I have no problem with the test existing but don't really like it being a part of the main build process; it makes compilation depend on an internet connection and, if there are enough such tests, would slow builds down significantly.